src/Document.php

```diff
@@ -707,6 +707,11 @@ public function canViewFile(array $options = [])
         $itemtype = $options['itemtype'] ?? null;
         $items_id = $options['items_id'] ?? null;
 
+        // Validate itemtype, as it might come directly from $_GET
+        if ($itemtype !== null && !Toolbox::isCommonDBTM($itemtype)) {
```

Comment:
IMHO, just adding a `&& is_a($itemtype, CommonDBTM::class, true)` in following block would probably be more easier to understand:
```diff
        if (
            $itemtype !== null
+           && is_a($itemtype, CommonDBTM::class, true)
            && $items_id !== null
            && $this->canViewFileFromItem($itemtype, $items_id)
        ) {
            return true;
        }
```